src/catalog/src/expr_cache.rs

```diff
@@ -57,57 +56,70 @@ struct CacheKey {
     id: GlobalId,
 }
 
-#[derive(Debug, Serialize, Deserialize)]
-struct CodecKey<'a> {
-    cache_key: Cow<'a, CacheKey>,
-    expressions: Cow<'a, Expressions>,
-}
-
 #[derive(Debug)]
 struct ExpressionCodec;
 
 impl DurableCacheCodec for ExpressionCodec {
     type Key = CacheKey;
-    type Val = Expressions;
+    // We use a raw JSON value instead of `Expressions` so that there is no backwards compatibility
+    // requirement on `Expressions` between versions.
+    type Val = serde_json::Value;
     type KeyCodec = SourceData;
-    type ValCodec = ();
+    type ValCodec = SourceData;
```

Comment:
Hmm, why? Seems basically as easy to use `<SourceData, (), ...` with a RelationDesc with both columns. That would keep the door open for exposing this via SQL

Reply:
Hmm, I think I totally misunderstood your comment on the last PR. Let's huddle today/tomorrow to discuss.

Reply:
Ah, yeah I probably just said something that was unclear. Happy to huddle!

Reply:
This is updated to use the SQL compatible schema.